### ISSUE TITLE
Termux-api hangs on Android 9

### ISSUE THREAD

**Issue by Dmole** [termux/termux-packages#3809]
https://github.com/termux/termux-packages/issues/334

Please remove the package or update the wiki 

https://wiki.termux.com/wiki/Termux:API

**Comment by Grimler91** [comment]
It works fine on my android 9 device. Have you installed the termux-api **app**? 

You need to provide more information if we are suppose to be able to help you.
* Which device is it?
* What commands have you tried?
* What does a log cat say?
* What's the output of `termux-info`?

**Comment by Dmole** [comment]
Sorry did not properly RTFM.

**Comment by AlainKnaff** [comment]
I have the same issue here. Both the termux-api package and Termux:API api app are installed. And yet, commands such as "termux-torch on" hang forever. Any ideas?
This happens on a Chinese-made S8_EEA tablet.
On my lineage OS phone everything works ok.

logcat -s termux:* termux-api:* just says "--------- beginning of main" "--------- beginning of crash"and nothing else...

**Comment by AlainKnaff** [comment]
termux-info also hangs for some time, but eventually says the following:

Packages CPU architecture:
arm
Subscribed repositories:
# sources.list
deb https://grimler.se/termux-packages-24/ stable main
# game-repo (sources.list.d/game.list)
deb https://dl.bintray.com/grimler/game-packages-24 games stable
# science-repo (sources.list.d/science.list)
deb https://dl.bintray.com/grimler/science-packages-24 science stable
Updatable packages:
All packages up to date
Android version:
9
Kernel build information:
Linux localhost 4.4.147 #4 SMP PREEMPT Mon Jun 22 11:16:28 CST 2020 armv7l Android
Device manufacturer:
incar
Device model:
S8_EEA

**Comment by Dmole** [comment]
A 3 second hang is normal;

```
time termux-info
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://termux.org/packages/ stable main
# game-repo (sources.list.d/game.list)
deb https://dl.bintray.com/grimler/game-packages-24 games stable
# science-repo (sources.list.d/science.list)
deb https://dl.bintray.com/grimler/science-packages-24 science stable
Updatable packages:
All packages up to date
Android version:
11
Kernel build information:
Linux localhost 4.4.223-g1bc566a21f60-ab6662096 #1 SMP PREEMPT Wed Jul 8 09:51:18 UTC 2020 aarch64 Android
Device manufacturer:
Google
Device model:
Pixel 2

real    0m3.920s
user    0m0.937s
sys     0m0.797s
```

**Comment by AlainKnaff** [comment]
Ok, I didn't know that by default, logcat only shows the application's own logs.
So I tried again from adb, and I got this:

10-27 23:57:36.249 27910 27910 W /system/bin/app_process: Could not reserve sentinel fault page
10-27 23:57:36.583 27910 27910 D AndroidRuntime: Calling main entry com.example.termuxam.Am
10-27 23:57:36.641   627  3120 D BroadcastQueue: Add broadcast <BroadcastRecord{df5cd65 u0 null}> into [ordered | background], pending size 0
10-27 23:57:36.641   627  3120 D BroadcastQueue: Header is BroadcastRecord{df5cd65 u0 null} now
10-27 23:57:36.641   627   642 D BroadcastQueue: Skipping delivery of ordered [background] BroadcastRecord{df5cd65 u0 null} for whatever reason
10-27 23:57:36.642   627   642 D BroadcastQueue: Finished with ordered broadcast BroadcastRecord{df5cd65 u0 null} receivers:(1) take 0ms in [background], remains 0
10-27 23:57:36.645 27910 27910 D AndroidRuntime: Shutting down VM
10-27 23:57:36.734   627   731 W ProcessCpuTracker: Skipping unknown process pid 27910

Of interest is the " BroadcastQueue: Skipping delivery of ordered [background] BroadcastRecord{df5cd65 u0 null} for whatever reason" line.

After testing to send broadcast intents to another application, I noticed that on this S8_EEA tablet, broadcast intents are only delivered if the application currently displays an activity. Just being installed, or even running a service is not enough. This is the case even for explicit intents which mention the target package and class.
As termux-api only has services and a broadcast receiver, it is never the case that it has an activity running, and so will never receive broadcast intents on this tablet.

Perusing the source code of BroadcastQueue.java (https://android.googlesource.com/platform/frameworks/base/+/kitkat-release/services/java/com/android/server/am/BroadcastQueue.java ... unfortunately I couldn't find this for current Android versions :-( ), I see that the only case where the "whatever reason" is not detailed is if the intent has been blocked by the "Intent Firewall". Could it be that Vankyo (the vendor of that S8_EEA tablet) has configured its "Intent Firewall" in an overly strict manner? Or maybe, since kitekat there have been other cases added where the "whatever reason" is not detailed.
